src/protocols/output_power_management.rs

```diff
@@ -0,0 +1,238 @@
+use std::collections::HashMap;
+
+use smithay::{
+    output::Output,
+    reexports::{
+        wayland_protocols_wlr::output_power_management::v1::server::{
+            zwlr_output_power_manager_v1, zwlr_output_power_v1,
+        },
+        wayland_server::{
+            Client, DataInit, Dispatch, DisplayHandle, GlobalDispatch, New, Resource,
+        },
+    },
+};
+use zwlr_output_power_manager_v1::ZwlrOutputPowerManagerV1;
+use zwlr_output_power_v1::ZwlrOutputPowerV1;
+
+const VERSION: u32 = 1;
+
+pub enum Mode {
+    Off,
+    On,
+}
+
+impl From<bool> for Mode {
+    fn from(value: bool) -> Self {
+        match value {
+            false => Self::Off,
+            true => Self::On,
+        }
+    }
+}
+
+impl From<zwlr_output_power_v1::Mode> for Mode {
+    fn from(value: zwlr_output_power_v1::Mode) -> Self {
+        match value {
+            zwlr_output_power_v1::Mode::Off => Self::Off,
+            zwlr_output_power_v1::Mode::On => Self::On,
+            _ => unreachable!(),
+        }
+    }
+}
+
+impl From<Mode> for zwlr_output_power_v1::Mode {
+    fn from(value: Mode) -> Self {
+        match value {
+            Mode::Off => Self::Off,
+            Mode::On => Self::On,
+        }
+    }
+}
+
+pub struct OutputPowerManagerGlobalData {
+    filter: Box<dyn for<'c> Fn(&'c Client) -> bool + Send + Sync>,
+}
+
+pub trait OutputPowerManagementHandler {
+    fn output_power_manager_state(&mut self) -> &mut OutputPowerManagementManagerState;
+
+    // todo: set output power per-monitor. Currently, niri is written to power
+    // all monitors at the same time.
```

Comment:
These callbacks on their own support it, so this comment is not needed here?

Reply:
Whoops, yeah